--- liblouisutdml-2.12.0/java/Jliblouisutdml.c.orig	2024-09-24 21:26:15.213409754 +0200
+++ liblouisutdml-2.12.0/java/Jliblouisutdml.c	2024-09-24 21:26:18.810098181 +0200
@@ -703,7 +703,7 @@ JNIEXPORT void JNICALL Java_org_liblouis
 }
 
 static jobject louLogCBFunc;
-static void javaLouLogCallback(int level, const char *message)
+static void javaLouLogCallback(logLevels level, const char *message)
 {
   execJavaLogCallback(louLogCBFunc, level, message);
 }
@@ -1340,7 +1340,7 @@ JNIEXPORT void JNICALL Java_org_liblouis
 }
 
 static jobject lbuLogCBFunc;
-static void javaLbuLogCallbackFunc(int level, const char *message)
+static void javaLbuLogCallbackFunc(logLevels level, const char *message)
 {
   execJavaLogCallback(lbuLogCBFunc, level, message);
 }
