--- liblouisutdml-2.6.0/liblouisutdml/logging.c.orig	2015-05-05 13:50:46.000000000 +0200
+++ liblouisutdml-2.6.0/liblouisutdml/logging.c	2016-07-30 17:44:06.067115720 +0200
@@ -60,7 +60,7 @@
   free(logMsg);
 }
 
-static void defaultLogCallback(int level, const char *message);
+static void defaultLogCallback(logLevels level, const char *message);
 
 static logcallback logCallbackFunction = defaultLogCallback;
 void EXPORT_CALL lbu_registerLogCallback(logcallback callback)
